cas-server-support-x509/src/main/java/org/jasig/cas/adaptors/x509/authentication/handler/support/LdapResourceCRLFetcher.java

```diff
@@ -0,0 +1,178 @@
+        this.securityAuthentication = securityAuthentication;
+    }
+
+    public void setSecurityPrincipal(final String securityPrincipal) {
+        this.securityPrincipal = securityPrincipal;
+    }
+
+    public void setSecurityCredentials(final String securityCredential) {
+        this.securityCredentials = securityCredential;
+    }
+
+    public void setObjectName(final String objectName) {
+        this.objectName = objectName;
+    }
+
+    @Override
+    protected X509CRL fetchInternal(final Object r) throws Exception {
+        if (r.toString().toLowerCase().startsWith("ldap")) {
+            return fetchCRLFromLdap(r);
+        }
+        return super.fetchInternal(r);
+    }
+
+    /**
+     * Downloads a CRL from given LDAP url, e.g.
+     * <code>ldap://ldap.infonotary.com/dc=identity-ca,dc=infonotary,dc=com</code>
+     *
+     * @param r the resource
+     * @return the x 509 cRL
+     * @throws Exception if connection to ldap fails, or attribute to get the revocation list is unavailable
+     */
+    protected X509CRL fetchCRLFromLdap(final Object r) throws Exception {
+        DirContext ctx = null;
+        try {
+            final String ldapURL = r.toString();
+            logger.debug("Fetching CRL from ldap {}", ldapURL);
+
+            final Map<String, String> env = configureLdapDirectoryContext(ldapURL);
+
+            logger.debug("Establishing a connection to {}", ldapURL);
+            ctx = new InitialDirContext((Hashtable) env);
+
+            logger.debug("Connected to {}", ldapURL);
+            if (StringUtils.isNotBlank(this.objectName)) {
+                logger.debug("Retrieving object {}", this.objectName);
+                ctx = (DirContext) ctx.lookup(this.objectName);
+            }
+
+            logger.debug("Retrieving certificate revocation list attribute {}",
+                    this.certificateRevocationListAttributeName);
+            final Attributes attrs = ctx.getAttributes("");
+            final Attribute aval = attrs.get(this.certificateRevocationListAttributeName);
+            if (aval != null) {
+                final byte[] val = (byte[]) aval.get();
+                if (val == null || val.length == 0) {
+                    throw new CertificateException("Empty attribute. Can not download CRL from: " + ldapURL);
+                }
+                final byte[] decoded64 = CompressionUtils.decodeBase64ToByteArray(val);
+                logger.debug("Retrieved CRL from ldap as byte array decoded in base64. Fetching...");
+                return super.fetch(new ByteArrayResource(decoded64));
+            }
+            throw new CertificateException("Attribute not found. Can not retrieve CRL from attribute: "
+                    + this.certificateRevocationListAttributeName);
+        } catch (final AuthenticationException | OperationNotSupportedException e) {
+            logger.error(e.getMessage(), e);
+            throw new CertificateException(e);
+        } finally {
```

Comment:
We could leverage try with resources pattern with an ldaptive `Connection` if you take my suggestion above.